### PR TITLE
C++ Solution + Brief Editorial for Snow Boots (Silver)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
new pr !!

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [ ] I have tested my code
- [ ] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [ ] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-811.mdx [hunk 1]
```diff
@@ -2,16 +2,87 @@
...
+#include <cstdio>
+#include <algorithm>
+
+using namespace std;
+
+int n;
+int m;
+int depths[251];
+bool vis[251][251];
+pair<int, int> shoes[251];
```
Comment (SansPapyrus683) [inline solutions/usaco-811.mdx:32]: 1. put in constant MAX_N
2. when declaring these arrays, use MAX_N + 1 (or be a normal person & use 0-indexing)
3. even better, use vectors

File: solutions/usaco-811.mdx [hunk 2]
```diff
@@ -2,16 +2,87 @@
...
+
+int ans = 1e9;
+
+void dfs(int depth, int boot)  {
+	// already visited this path
+	if (vis[depth][boot]) return;
```
Comment (SansPapyrus683) [inline solutions/usaco-811.mdx:38]: braces

File: solutions/usaco-811.mdx [hunk 3]
```diff
@@ -2,16 +2,87 @@
...
+
+using namespace std;
+
+int n;
+int m;
+int depths[251];
+bool vis[251][251];
+pair<int, int> shoes[251];
+
+int ans = 1e9;
```
Comment (SansPapyrus683) [inline solutions/usaco-811.mdx:34]: INT32_MAX

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
all these globals are necessary right

File: solutions/usaco-811.mdx [hunk 4]
```diff
@@ -25,18 +25,21 @@ Due to the low bounds on $N$ and $B$, we can run a [DFS (Depth First Search)](/s
...
 using namespace std;
 
+const int MAX_N = 250;
 int n;
 int m;
-int depths[251];
-bool vis[251][251];
-pair<int, int> shoes[251];
+vector<int> depths(MAX_N + 1);
+bool vis[MAX_N + 1][MAX_N + 1];
```
Comment (SansPapyrus683) [inline solutions/usaco-811.mdx:32]: uuuuh if you made the others vectors why not this one?

Comment (ryanchou-dev) [reply solutions/usaco-811.mdx:32]: vectors are weird for 2d arrays

Comment (ryanchou-dev) [reply solutions/usaco-811.mdx:32]: also i normally do a bool array for `visited`. if you want me to change it, ill do it :D

Comment (SansPapyrus683) [reply solutions/usaco-811.mdx:32]: really?
it's just `vector<vector<bool>> x(size, vector<bool>(size));`

Comment (ryanchou-dev) [reply solutions/usaco-811.mdx:32]: alr, ill do it after aoc

**Comment by SansPapyrus683** [comment]
does the code still run

**Comment by ryanchou-dev** [comment]
oh shoot, forgot to include vector

**Comment by ryanchou-dev** [comment]
> does the code still run

yep

**Review by SansPapyrus683 — APPROVED** [review]
ily so much ryan

**Comment by SansPapyrus683** [comment]
also is there a reason for the vectors to be 1-indexed?

**Comment by ryanchou-dev** [comment]
> also is there a reason for the vectors to be 1-indexed?

nope, ill make them zero-indexed

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**